akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -941,35 +940,69 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
-          // If it's time to try to gossip to some nodes with a different view
-          // gossip to a random alive member with preference to a member with older gossip version
-          localGossip.members.collect {
-            case m if !localGossip.seenByNode(m.uniqueAddress) && validNodeForGossip(m.uniqueAddress) ⇒
-              m.uniqueAddress
-          }(breakOut)
-        } else Vector.empty
-
-      if (preferredGossipTargets.nonEmpty) {
-        val peer = selectRandomNode(preferredGossipTargets)
-        // send full gossip because it has different view
-        peer foreach gossipTo
-      } else {
-        // Fall back to localGossip; important to not accidentally use `map` of the SortedSet, since the original order is not preserved)
-        val peer = selectRandomNode(localGossip.members.toIndexedSeq.collect {
-          case m if validNodeForGossip(m.uniqueAddress) ⇒ m.uniqueAddress
-        })
-        peer foreach { node ⇒
-          if (localGossip.seenByNode(node)) gossipStatusTo(node)
-          else gossipTo(node)
-        }
+  /**
+   * Chooses a set of possible gossip targets that is in the same dc. If the cluster is not multi dc this will
+   * means it is a choice among all nodes of the cluster.
+   */
+  def localDcGossipTargets(): Vector[UniqueAddress] = {
+    val firstSelection =
+      if (ThreadLocalRandom.current.nextDouble() < adjustedGossipDifferentViewProbability) {
+        // If it's time to try to gossip to some nodes with a different view
+        // gossip to a random alive same dc member with preference to a member with older gossip version
+        latestGossip.members.collect {
+          case m if m.team == selfTeam && !latestGossip.seenByNode(m.uniqueAddress) && validNodeForGossip(m.uniqueAddress) ⇒
+            m.uniqueAddress
+        }(breakOut)
+      } else Vector.empty
+
+    // Fall back to localGossip
+    if (firstSelection.isEmpty) {
+      latestGossip.members.toVector.collect {
+        case m if m.team == selfTeam && validNodeForGossip(m.uniqueAddress) ⇒ m.uniqueAddress
       }
+    } else Vector.empty
+
+  }
+
+  /**
+   * Choose cross-dc nodes if this one of the N oldest nodes, and if not fall back to gosip locally in the dc
+   */
+  def multiDcGossipTargets(): Vector[UniqueAddress] = {
+    // TODO cache this until gossip changes??
+    val nodesPerDc = latestGossip.members.foldLeft(Map.empty[String, SortedSet[Member]]) { (acc, member) ⇒
+      acc.get(member.team) match {
+        case Some(set) if set.size > cluster.settings.CrossDcConnections ⇒ acc
+        case Some(set) ⇒ acc + (member.team → (set + member))
+        case None ⇒ acc + (member.team → (SortedSet.empty(Member.ageOrdering) + member))
+      }
+    }
+
+    val localTargets = localDcGossipTargets()
+    // only do cross DC gossip if this node is among the N oldest
+    if (!nodesPerDc(selfTeam).exists(_.uniqueAddress == selfUniqueAddress)) localTargets
```

Comment:
`!nodesPerDc(selfTeam).contains(selfMember)`

but we don't have a `selfMember`, I might introduce that when I do the `Gossip` refactoring